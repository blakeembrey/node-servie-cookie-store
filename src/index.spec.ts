import Keygrip = require('keygrip')
import { Request, Response } from 'servie'
import { Cookie } from './index'

describe('servie cookie store', () => {
  it('should work with regular cookies', () => {
    const req = new Request({ url: '/' })
    const res = new Response()
    const cookie = new Cookie(req, res)

    req.headers.set('Cookie', `a=${cookie.encode(1)}; b=${cookie.encode(2)}`)

    expect(cookie.get('a')).toEqual(1)
    expect(cookie.get('b')).toEqual(2)

    cookie.set('a', { test: true })
    cookie.set('b', 'abc')

    expect(res.headers.object()).toMatchSnapshot()
  })

  it('should sign cookies', () => {
    const req = new Request({ url: '/' })
    const res = new Response()
    const cookie = new Cookie(req, res, new Keygrip(['secret']))

    req.headers.set('Cookie', `a=${cookie.encode(1)}; b=${cookie.encode(2)}`)

    expect(cookie.get('a')).toEqual(1)
    expect(cookie.get('b')).toEqual(2)

    cookie.set('a', { test: true })
    cookie.set('b', 'abc')

    expect(res.headers.object()).toMatchSnapshot()
  })

  it('should ignore bad decodes', () => {
    const req = new Request({ url: '/' })
    const res = new Response()
    const cookie = new Cookie(req, res, new Keygrip(['secret']))

    req.headers.set('Cookie', `a=1; b=${cookie.encode(2).split('.')[0]}.fakedigest`)

    expect(cookie.get('a')).toBe(undefined)
    expect(cookie.get('b')).toBe(undefined)
  })
})
