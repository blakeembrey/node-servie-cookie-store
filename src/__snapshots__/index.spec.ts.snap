// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`servie cookie store should sign cookies 1`] = `
Object {
  "set-cookie": Array [
    "a=eyJ0ZXN0Ijp0cnVlfQ%3D%3D._6w9E-9zKPwThSnsjUD9fJY5RHs",
    "b=ImFiYyI%3D.sxQLgTnZKd6G9fXWj161wzF6yEA",
  ],
}
`;

exports[`servie cookie store should work with regular cookies 1`] = `
Object {
  "set-cookie": Array [
    "a=eyJ0ZXN0Ijp0cnVlfQ%3D%3D",
    "b=ImFiYyI%3D",
  ],
}
`;
